tasks/main.yml

```diff
@@ -1,25 +1,110 @@
 # SPDX-License-Identifier: MIT
 ---
-- name: Set platform/version specific variables
-  include_tasks: tasks/set_vars.yml
-
-# Examples of some tasks:
 - name: Ensure required packages are installed
   package:
-    name: "{{ __template_packages }}"
+    name: '{{ __systemd_packages }}'
     state: present
 
-- name: Ensure required services are enabled and started
-  service:
-    name: "{{ item }}"
+- name: Deploy required unit files
+  template:
+    src: '{{ item }}.j2'
+    dest: '{{ __admin_units_dir }}/{{ item }}'
+    owner: root
+    group: root
+    mode: '0644'
+  when: systemd_unit_files is defined
+  with_items:
+    - '{{ systemd_unit_files }}'
+
+- name: Create drop-in directories
+  file:
+    path: '{{ __admin_units_dir }}/{{ ".".join(item.split(".")[:-1]) }}.d/'
+    state: directory
+    owner: root
+    group: root
+    mode: '0755'
+  when: systemd_dropins is defined
+  with_items:
+    - '{{ systemd_dropins }}'
+
+- name: Deploy systemd drop in config
+  template:
+    src: '{{ item }}.j2'
+    dest: "{{ __admin_units_dir }}/\
+     {{ \".\".join(item.split(\".\")[:-1]) }}.d/\
+     99-override.conf"
+    owner: root
+    group: root
+    mode: '0644'
+  when: systemd_dropins is defined
+  with_items:
+    - '{{ systemd_dropins }}'
+
+- name: Reload systemd
+  systemd:
+    daemon_reload: true
+  when: systemd_unit_files is defined or systemd_dropins is defined
```

Comment:
This should really trigger only when something has *changed* - if you run the role twice with the exact same `systemd_unit_files` and `systemd_dropins`, then you will reload systemd even when you did not want to, and the role is not idempotent.  The way to do this is to use `register` to register the changes from managing the unit files and dropins e.g.

```yaml
- name: Deploy required unit files
...
register: __systemd_unit_files_result

- name: Deploy systemd drop in config
...
register: __systemd_dropin_result

- name: Reload systemd
  systemd:
    daemon_reload: true
  when: __systemd_unit_files_result is changed or __systemd_dropin_result is changed
```